### PR TITLE
fix(3d-tiles): Fix implicit tileset subtree parsing

### PR DESCRIPTION
**Pull request by Kaapp**
Fixes https://github.com/visgl/loaders.gl/issues/2820

A couple of minor bugs in the parsing algorithm for implicit tiles which would cause it to fail when processing tilesets with multiple subtrees:
- A pass-by-ref bug with `globalData` in the recursive parsing algorithm which would lead to calculating tile and morton indices incorrectly once we reached the maximum depth of the tree and attempted to evaluate a second subtree
- Shifting the global tile coordinate values too far leading to requesting the wrong tile content URIs, it only needs to be one bit per coordinate as we're calculating x/y/z individually.

For a sparse tileset, empty tiles would have requests made to the baseUrl (which will always fail) due to default behaviour in `resolveUri`, so I've put in a check for that.

Lastly, and the majority of the PR diff, the original test file had `contentAvailability` which did not match the content structure beyond the first few levels, so I've replaced it with one of Cesium's demo files and increased the depth of the test to ensure we query multiple subtrees (i believe Cesium's test file is under the same Apache 2.0 licence based on their [contributor licence agreement](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md))

### CODE REVIEW

**Comment by Kaapp** [comment]
@belom88 yep, it's a bit of a beast to unpack. specifically i believe [this part](https://github.com/CesiumGS/3d-tiles/blob/main/specification/ImplicitTiling/AVAILABILITY.adoc#implicittiling-availability-indexing:~:text=(x%2C%20y%2C%20z)%20coordinates%20follow,globalZ%20%3D%20concatBits(subtreeRoot.globalZ%2C%20tile.localZ)) was the main cause where global tile coordinates were being shifted too many times:
> (x, y, z) coordinates follow the same pattern as Morton indices. The only difference is that the concatenation of bits happens component-wise. That is:
> 
> ``` 
> tile.globalX = concatBits(subtreeRoot.globalX, tile.localX)
> tile.globalY = concatBits(subtreeRoot.globalY, tile.localY)
> 
> // Octrees only
> tile.globalZ = concatBits(subtreeRoot.globalZ, tile.localZ)
> ```

The second part was just that the original `globalData` was being mutated rather than a new object reference being supplied to the recursion, so when the recursion finished evaluation a subtree it wouldn't reset the `globalData` values back to the parent subtree and would use the mutated values for the second and subsequent subtrees

**Comment by ibgreen** [comment]
@Kaapp Do you need help with making this PR pass CI?

File: modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.ts [hunk 1]
```diff
@@ -80,6 +80,12 @@ function getRefine(refine?: string): TILE_REFINEMENT | string | undefined {
 }
 
 function resolveUri(uri: string = '', basePath: string): string {
+  if (uri === '') {
```
[the remark this answers is not in the document]
Comment (Kaapp) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.ts]: Have moved the check to the call that matters, although the default value still worries me a bit for someone else falling into this trap - but perhaps something to look at another time.

[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.ts]: To cover up this trap, we could consider throwing an error if input string is empty. 

Just silently returning an empty string in this case feels like setting up another trap.

**Comment by Kaapp** [comment]
@ibgreen apologies for the delay as i had been away - i can't trigger workflows but the last commit showed a single test failure in the `core` package which is a little strange as this PR doesn't touch that. wasn't sure if there was some flaky behaviour in it but it doesn't seem to appear locally for me with `yarn test ci`

File: modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.ts]: Yes, sounds good!

**Comment by Kaapp** [comment]
all good now 😃

**Pull request merged**